src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +193,135 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             }
         }));
     }
+
+    private async Task WatchDashboardLogsAsync()
+    {
+        async Task<string?> GetDashboardResourceIdentifier()
+        {
+            string? dashboardResourceId = null;
+
+            try
+            {
+                var timeout = Debugger.IsAttached ? Timeout.InfiniteTimeSpan : dashboardOptions.Value.DashboardStartupTimeout;
+
+                // The dashboard resource isn't immediately available so watch for it.
+                // Wait for dashboard to startup and be reported before giving up.
+                using var cts = new CancellationTokenSource(timeout);
+
+                await foreach (var notification in resourceNotificationService.WatchAsync(cts.Token))
+                {
+                    // We want to capture the replica of the dashboard resource. This will allow us to get the logs for the dashboard.
+                    // When the resource id is not the same as the resource name, it's the replica.
+                    if (StringComparers.ResourceName.Equals(notification.Resource.Name, KnownResourceNames.AspireDashboard) &&
+                        notification.ResourceId != notification.Resource.Name)
+                    {
+                        dashboardResourceId = notification.ResourceId;
+                        break;
+                    }
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                if (distributedApplicationLogger.IsEnabled(LogLevel.Debug))
+                {
+                    distributedApplicationLogger.LogDebug("Timed out waiting for dashboard resource to start.");
+                }
+            }
+
+            return dashboardResourceId;
+        }
+
+        if (await GetDashboardResourceIdentifier().ConfigureAwait(false) is not string dashboardResourceId)
+        {
+            // The only way this can happen is if DCP is unhealthy, so log a warning.
+            distributedApplicationLogger.LogWarning("Failed to get the dashboard logs.");
+            return;
+        }
+
+        var loggerCache = new Dictionary<string, ILogger>(StringComparer.Ordinal);
+        var defaultDashboardLogger = loggerFactory.CreateLogger("Aspire.Hosting.Dashboard");
+
+        try
+        {
+            // We turned on the JSON formatter for the logger, so we can log the log lines as JSON.
+            await foreach (var batch in resourceLoggerService.WatchAsync(dashboardResourceId).WithCancellation(_shutdownCts.Token).ConfigureAwait(false))
```

Comment:
There is a call to dispose the CTS in DisposeAsync: `_shutdownCts.Dispose();`

It's possible that dispose happens before the token property on the CTS here is accessed. An ObjectDisposedException is thrown from `CancellationTokenSource.Token`.

Potential fixes:
* Pass the token from the CTS into the method as a parameter and then always use the parameter <- best imo
* Don't dispose _shutdownCts

Reply:
Never dispose the cts, that's the rule 😄 . Done